src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8039,7 +8095,9 @@ void reportBadDelegateParameter(BindingDiagnosticBag bag, MethodSymbol sourceInv
                         canConvertNestedNullability = conversion.Exists;
                     }
 
-                    resultState = conversion.IsReference ? getReferenceConversionResultState(targetTypeWithNullability, operandType) : operandType.State;
+                    resultState = conversion.IsReference
+                        ? getReferenceConversionResultState(targetTypeWithNullability, operandType)
+                        : (operandType.State == NullableFlowState.MaybeDefault && GetPlaceholderStateOrMaybeDefault(conversionOperand) == NullableFlowState.NotNull ? NullableFlowState.NotNull : operandType.State);
```

Comment:
I'm not certain whether this is the right place to do this. I'll probably need to check out your version and step through it.